ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -159,6 +159,197 @@ objects:
+            curl -s -L ${DOWNLOAD_URL}/${ETCD_VER}/etcd-${ETCD_VER}-linux-amd64.tar.gz -o $ASSET_DIR/tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz \
+              && tar -xzf $ASSET_DIR/tmp/etcd-${ETCD_VER}-linux-amd64.tar.gz -C $ASSET_DIR/shared --strip-components=1 \
+              && mv $ASSET_DIR/shared/etcdctl $ASSET_DIR/bin/ \
+              && rm $ASSET_DIR/shared/etcd \
+              && ETCDCTL_API=3 $ASSET_DIR/bin/etcdctl version
+
+            ETCD_NAME="etcd-member-$(hostname -f)"
+
+            rm -rf /var/lib/etcd/
+
+            source /run/etcd/environment
+            ETCDCTL_API=3 etcdctl \
+              --cert /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.crt \
+              --key /etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.key \
+              --cacert /etc/ssl/etcd/ca.crt \
+              snapshot restore /tmp/snapshot.db \
+              --name ${ETCD_NAME} \
+              --initial-cluster "$@" \
+              --initial-cluster-token etcd-cluster-1 \
+              --skip-hash-check=true \
+              --initial-advertise-peer-urls https://${ETCD_IPV4_ADDRESS}:2380 \
+              --data-dir /var/lib/etcd/
+        EOF
+          chmod +x /tmp/etcd_restore.sh
+          for master in "${MASTERS[@]}"
+          do
+            scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" /tmp/etcd_restore.sh "core@${master}":/tmp
+            bastion_ssh "core@${master}" "sudo -i /tmp/etcd_restore.sh ${ETCD_CONN_STRING}"
+          done
+
+          for master in "${MASTERS[@]}"
+          do
+            while true
+            do
+              bastion_ssh "core@${master}" "sudo -i mv /etc/kubernetes/manifests-backup/* /etc/kubernetes/manifests" && break; sleep 10
+            done
+          done
+
+          echo "Wait for API server to come up"
+          for i in $(seq 0 10)
+          do
+            oc get nodes && break
+            sleep 30
+          done
+
+          echo "Wait for MCO to rollout new configs"
+          for i in $(seq 0 10); do oc get machineconfigpool/master > /dev/null && break; sleep 30; done
+          wait_for_machineconfigpool_to_apply
+
+          echo "Verify 99-rollback-test machineconfig"
+          MC="$(oc get machineconfig/99-rollback-test -o jsonpath='{.spec.config.storage.files[0].contents.source}')"
+          if [[ "${MC}" != "data:,A" ]]; then
+            echo "Unexpected MachineConfig output: ${MC}"
+            exit 1
+          fi
+
+          echo "Verify /etc/rollback-test contents"
+          rc=0
+          for master in "${MASTERS[@]}"
+          do
+            ROLLBACK="$(bastion_ssh core@${master} "sudo -i cat /etc/rollback-test")"
+            if [[ "${ROLLBACK}" != "A" ]]; then
+              echo "Master ${master} rollback contents was ${ROLLBACK}"
+              rc=1
+            fi
+          done
+
+          if [[ "${rc}" == "1" ]]; then exit 1; fi
```

Comment:
I think we want `exit 1` -> `return 1`, so the caller can decide if they want to add additional error handling.

Reply:
never mind, the rest of this script is all `set -e`, so this matches that.